pkg/applicationmanager/controllers/applicationbackup.go

```diff
@@ -262,6 +262,46 @@ func (a *ApplicationBackupController) handle(ctx context.Context, backup *stork_
 			if !canDelete {
 				return nil
 			}
+			// run the post exec rules if the backup is in volume stage and execRulesCompleted is not completed
+			if _, ok := a.execRulesCompleted[string(backup.UID)]; !ok && backup.Status.Stage == stork_api.ApplicationBackupStageVolumes {
+				if terminationChannels, ok := a.terminationChannels[string(backup.UID)]; ok {
+					for _, channel := range terminationChannels {
+						logrus.Infof("Sending termination commands to kill pre-exec pod in non-kdmp driver path")
+						channel <- true
+					}
+				}
+				if backup.Spec.PostExecRule != "" {
+					log.ApplicationBackupLog(backup).Infof("Starting post-exec rule in delete path")
+					err = a.runPostExecRule(backup)
+					if err != nil {
+						message := fmt.Sprintf("Error running PostExecRule: %v", err)
+						log.ApplicationBackupLog(backup).Errorf(message)
+						a.recorder.Event(backup,
+							v1.EventTypeWarning,
+							string(stork_api.ApplicationBackupStatusFailed),
+							message)
+						return fmt.Errorf("%v", message)
+					}
+					a.execRulesCompleted[string(backup.UID)] = true
+				}
+			}
+
+			// Delete post-exec rule CR if any
+			if len(backup.Spec.PostExecRule) != 0 && backup.Annotations[utils.PxbackupAnnotationCreateByKey] == utils.PxbackupAnnotationCreateByValue && backup.Annotations["portworx.io/backup-by"] != "backup-schedule" && backup.Annotations["portworx.io/backup-by"] != "restore" {
```

Comment:
Good check. Can we please some comment, why we are having schedule and restore check? So that it will be useful for  others.

Also, We might need to avoid the running the post-Exec rule, if it created  by restore case rt?

Reply:
1. Sure will update the comments.
2. Also, We might need to avoid the running the post-Exec rule, if it created by restore case rt? - Yes, existing check will prevent this (**backup.Annotations["portworx.io/backup-by"] != "restore"**) is there anything else am missing ?

Reply:
Ok, I meant of avoiding the running post-exec rule if the applicationbackup CR was created by restore operation.
But I think the check in the line 254 might not satisfy for restore's applicationbakup CR. So it should be fine.

Reply:
One concern is that when the application backup CR is deleted, px-backup will also delete the Rule CR as well rt?
Will it not cause a issue that Rule CR might be available at line 264?

Reply:
A PR is also raised for the same https://github.com/portworx/px-backup/pull/1432